### PR TITLE
Fix to PrimaryProtonGun event generator module to remove 'R' designation

### PR DESCRIPTION
**Pull request by goodenou**
Removed 'R' designation from module, and added FHICL validation.

### CODE REVIEW

**Comment by kutschke** [comment]
What tests have you run?   Is this pull request expected to change the results of nightly validation?

**Comment by goodenou** [comment]
I have not run any validation, but I can it over any relevant configurations.  I believe Validation/fcl/potSim.fcl might be the only related validation config file.  Am I correct?  I have run all of the config files that referenced this module, and they all run with issue.

**Comment by kutschke** [comment]
HI Lisa,

    I believe you have checked that it compiles and that it runs all of the fcl files that you modified.  But you only ran a few events to verify that it run.  Right?

    Run the nightly pot test.  If that looks good we are ready to go.

Rob

**Comment by goodenou** [comment]
Yes, I checked that it compiles and ran over all of the fcl files that use that generator.  I just added in the modifications to a few more fcl files, in particular the one Ray has been wanting fixed, JobConfig/validation/stoppedMuonsSingleStage.fcl.  I have checked that if one of the required fcl params is missing, that the job fails.  I do not know what the nightly POT test is.  I will ask Ray.

File: EventGenerator/inc/PrimaryProtonGunImpl.hh [hunk 1]
```diff
@@ -0,0 +1,140 @@
+#ifndef EventGenerator_PrimaryProtonGunImpl_hh
+#define EventGenerator_PrimaryProtonGunImpl_hh
+//
+// Generate a proton with the primary proton energy
+//
+// $Id: PrimaryProtonGunImpl.hh,v 1.17 2013/12/13 21:35:07 gandr Exp $
+// $Author: gandr $
+// $Date: 2013/12/13 21:35:07 $
```
Comment (kutschke) [inline EventGenerator/inc/PrimaryProtonGunImpl.hh]: When you encounter these comments - just delete them.  Please do this as targets of opportunity.  They are left overs from a version control system, cvs, that we stopped using a long time ago.  With git they add no value - just noise.

File: EventGenerator/fcl/prolog.fcl [hunk 2]
```diff
@@ -108,13 +121,6 @@ StoppedParticleReactionGun: {
 }
 
 
-StoppedParticleReactionGunN: {
-    @table::StoppedParticleReactionGun
-    module_type    : StoppedParticleReactionGunN
-    stashSize      : @nil
-}
```
Comment (kutschke) [inline EventGenerator/fcl/prolog.fcl]: I presume that StoppedParticleReactionGunN was removed in an earlier commit and this is just left over cleanup?

**Comment by resnegfk** [comment]
Small comments/questions:

I still see some "...GunN" in: 
Analyses/test/genRecoMT.fcl
EventGenerator/test/conversionN.fcl
EventGenerator/test/protonGunN.fcl
EventGenerator/test/reactionN.fcl
are those needed?

Is explicit needed in:
explicit PrimaryProtonGunImpl(CLHEP::HepRandomEngine& engine, const PrimaryProtonGunConfig& config);
Is there a conversion you are trying to avoid?

File: EventGenerator/fcl/prolog.fcl (hunk 2 above)
Comment (goodenou) [reply EventGenerator/fcl/prolog.fcl]: Yes, that is correct.

**Comment by goodenou** [comment]
Thanks for pointing out those 'N' files.  I will remove those.

Regarding the form of the constructor, I am following the instructions shown here:
https://cdcvs.fnal.gov/redmine/projects/art/wiki/Configuration_validation_and_description

**Comment by goodenou** [comment]
I am in the process of doing the validation test now.

**Comment by goodenou** [comment]
I ran validation on the potSim.fcl configuration.  There was perfect agreement on all 116 plots.

**Comment by goodenou** [comment]
The deprecated "N" fcl files have now been removed.

**Comment by kutschke** [comment]
Thanks Lisa,

  I would  like to hear from Ray and Krzysztof.   Then I am ready to merge.

Rob

**Comment by kutschke** [comment]
Thanks Everyone,

  I merged the pull request.  

Rob

**Pull request merged**